app/vmalert/manager.go

```diff
@@ -97,19 +98,36 @@ func (m *manager) update(ctx context.Context, path []string, validateTpl, valida
 	for _, og := range m.groups {
 		ng, ok := groupsRegistry[og.ID()]
 		if !ok {
+			startAt := time.Now()
 			// old group is not present in new list
 			// and must be stopped and deleted
 			og.close()
 			delete(m.groups, og.ID())
+			logger.Infof("group %q delete success, took %v seconds", og.Name, time.Since(startAt).Seconds())
 			og = nil
 			continue
 		}
-		og.updateCh <- ng
+		go func(og *Group, ng *Group) {
+			startAt := time.Now()
+			select {
+			case <-ctx.Done():
+				logger.Infof("group %q: context cancelled", og.Name)
+				return
+			case <-og.doneCh:
+				logger.Infof("group %q: received stop signal", og.Name)
```

Comment:
this will be printed twice (here and in group goroutine) on group.Close since underlying channel will be closed